modules/ray_tracing/test/src/userobjects/RepeatableRayStudyBaseTest.C

```diff
@@ -0,0 +1,89 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "RepeatableRayStudyBaseTest.h"
+
+registerMooseObject("RayTracingTestApp", RepeatableRayStudyBaseTest);
+
+InputParameters
+RepeatableRayStudyBaseTest::validParams()
+{
+  auto params = RepeatableRayStudy::validParams();
+
+  params.addParam<bool>("create_non_unique_id_rays",
+                        "Create Rays on each processor with a non-unique ID");
+
+  params.addParam<processor_id_type>("create_additional_ray_pid",
+                                     "Create an additional Ray on the given processor");
+  params.addParam<processor_id_type>(
+      "create_non_replicated_id_ray_pid",
+      "Create a Ray with a non-replicated ID on the given processor");
+  params.addParam<processor_id_type>("create_non_replicated_ray_pid",
+                                     "Create a non-replicated Ray on the given processor");
+
+  params.suppressParameter<std::vector<Real>>("max_distances");
+  params.suppressParameter<std::vector<Point>>("end_points");
+
+  return params;
+}
+
+RepeatableRayStudyBaseTest::RepeatableRayStudyBaseTest(const InputParameters & parameters)
+  : RepeatableRayStudy(parameters)
+{
+}
+
+void
+RepeatableRayStudyBaseTest::defineRays()
+{
+  RepeatableRayStudy::defineRays();
+
+  if (parameters().isParamSetByUser("create_non_unique_id_rays") &&
+      getParam<bool>("create_non_unique_id_rays"))
+    _rays.push_back(acquireCopiedRay(*_rays[0]));
+
+  if (parameters().isParamSetByUser("create_additional_ray_pid"))
+  {
+    const auto pid = getParam<processor_id_type>("create_additional_ray_pid");
+
+    mooseAssert(pid < n_processors(), "Invalid pid");
+    auto ray = acquireRegisteredRay("additional_ray");
+
+    if (pid == _pid)
+    {
+      ray->setStart(_rays[0]->currentPoint());
+      ray->setStartingDirection(_rays[0]->direction());
+      _rays.push_back(ray);
+    }
+  }
+
+  if (parameters().isParamSetByUser("create_non_replicated_id_ray_pid"))
```

Comment:
check for param value

Reply:
Not needed. this is done below in an assertion (it's a test object)